internal/git/gitsync_processor.go

```diff
@@ -137,21 +137,21 @@ func watchRepo(ctx context.Context, r *git.Repository, gitSync *v1alpha1.GitSync
 			return err
 		}
 
-		err = updateCommitStatus(ctx, kubeClient, namespacedName, hash.String(), repo, logger)
+		err = updateCommitStatus(ctx, kubeClient, namespacedName, hash.String(), logger)
 		if err != nil {
 			return err
 		}
 
 		lastCommitHash = hash.String()
 	} else {
-		lastCommitHash = gitSync.Status.CommitStatus[repo.Name].Hash
+		lastCommitHash = gitSync.Status.CommitStatus.Hash
 	}
 	// no monitoring if targetRevision is a specific commit hash
 	if isCommitSHA(repo.TargetRevision) {
-		if val.Hash != hash.String() {
+		if gitSync.Status.CommitStatus.Hash != hash.String() {
```

Comment:
I know this was in here before, but I wonder if it's supposed to say `if lastCommitHash != hash.String()`? (for the case that there is nothing stored in the Status previously?)

Reply:
Yes, you're right. It is more appropriate to use `lastCommitHash` here.